Examples/Liquidate/BorrowSystem.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: agpl-3.0
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity ^0.8.0;
+
+interface IERC20 {
+	function transfer(address to, uint value) external;
+	function transferFrom(address from, address to, uint value) external;
+}
+
+interface IOracle {
+	function get(IERC20 token1, IERC20 token2) external returns (uint256);
+}
+
+contract SimpleBorrowSystem {
+	// oracle to be used
+	IOracle oracle;
+	// ERC20 collateralToken
+	IERC20 collateralToken;
+	// ERC20 borrowToken 
+	IERC20 borrowToken;
+
+	// amount of borrowToken user borrowed
+	mapping(address => uint256) public userBorrowAmount;
+	// amount of collateralToken user has deposited
+	mapping(address => uint256) public userCollateralAmount;
+
+	// actions for the batchCalls
+	uint8 internal constant CALL_BORROW = 1;
+	uint8 internal constant CALL_REPAY = 2;
+	uint8 internal constant CALL_LIQUIDATE = 3;
+
+	/* Borrow amount and provide collateral */
+	function borrow(uint256 borrowAmount, uint256 collateralAmount) public {
+		userBorrowAmount[msg.sender] += borrowAmount;
+		userCollateralAmount[msg.sender] += collateralAmount;
+
+		require(_isSolvent(msg.sender), "user is not solvent");
+
+		borrowToken.transfer(msg.sender, borrowAmount);
+		collateralToken.transferFrom(msg.sender, address(this), collateralAmount);
+	}
+
+	/* Repay part of a user's borrow and get back a part of his collateral */
+	function repay(uint256 borrowAmount, uint256 collateralAmount) public {
```

Comment:
You can choose how much `borrow` you pay and how much collateral you get in return?
Also it is not 'a user's borrow', it is 'A user repays a part of its borrow'. You cannot pay the borrow of other users.

Reply:
yes, as long as you are in solvent

Reply:
Can you change the description?
`The user (msg.sender) repays a part of its borrow and gets back a part of his collateral`